nixos/modules/config/users-groups.nix

```diff
@@ -92,7 +92,7 @@ let
 
       group = mkOption {
         type = types.str;
-        apply = x: assert (builtins.stringLength x < 17 || abort "Group name '${x}' is longer than 16 characters which is not allowed!"); x;
+        apply = x: assert (builtins.stringLength x < 32 || abort "Group name '${x}' is longer than 31 characters which is not allowed!"); x;
```

Comment:
Nitpick: are you sure +-1 is correct here?  `--with-group-name-max-length=32` would feel like 32 characters is still OK, but I didn't verify that at all.

Reply:
Ah, now I read your description in full so I see :-)